am-lib/src/main/java/uk/gov/hmcts/reform/amlib/models/DefaultPermissionGrant.java

```diff
@@ -17,12 +17,8 @@
 @Builder
 @SuppressWarnings("LineLength")
 public final class DefaultPermissionGrant {
-    @NotBlank
-    private final String serviceName;
-    @NotBlank
-    private final String resourceType;
-    @NotBlank
-    private final String resourceName;
+    @NotNull
+    private final ResourceDefinition definition;
```

Comment:
Mark it with `@Valid` annotation please.

Reply:
Sadly our tests did not spot that because `InvalidArgumentsProvider` does support nested complex objects at the moment.